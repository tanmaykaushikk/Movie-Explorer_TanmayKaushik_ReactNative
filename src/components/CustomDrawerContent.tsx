import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet, Dimensions, Image } from 'react-native';
import { DrawerContentComponentProps, DrawerContentScrollView } from '@react-navigation/drawer';
import { LinearGradient } from 'react-native-linear-gradient';

const { width, height } = Dimensions.get('window');

const wp = (percent:number) => (width * percent) / 100;
const hp = (percent:number) => (height * percent) / 100;

const CustomDrawerContent :React.FC<DrawerContentComponentProps>= (props) => {
  const { navigation } = props;
  
  const menuItems = [
    { name: 'Login', route: 'LoginPage', icon: 'üîë' },
    { name: 'Sign Up', route: 'SignupPage', icon: '‚ú®' },
    { name: 'Premium', route: 'Premium', icon: '‚≠ê' },
    { name: 'Profile', route: 'Profile', icon: 'üë§' },
  ];

  return (
    <LinearGradient
      colors={['#333333', '#222222', '#111111', '#000000']}
      start={{x: 0, y: 0}}
      end={{x: 1, y: 1}}
      style={styles.gradient}
    >
      <DrawerContentScrollView {...props} contentContainerStyle={styles.scrollView}>
        <View style={styles.header}>
          <View style={styles.logoContainer}>
            <View style={styles.logoBackground}>
              <Text style={styles.logoText}>A</Text>
            </View>
          </View>
          <Text style={styles.appName}>Movie Explorer</Text>
          <Text style={styles.tagline}>Deep-Dive into Movies</Text>
        </View>
        
        <View style={styles.divider} />
        
        <View style={styles.menuContainer}>
          {menuItems.map((item, index) => (
            <TouchableOpacity
              key={index}
              style={styles.menuItem}
              onPress={() => navigation.navigate(item.route)}
            >
              <View style={styles.iconContainer}>
                <Text style={styles.icon}>{item.icon}</Text>
              </View>
              <Text style={styles.menuText}>{item.name}</Text>
            </TouchableOpacity>
          ))}
        </View>
        
        <View style={styles.footer}>
          <TouchableOpacity 
            style={styles.footerButton}
            onPress={() => navigation.navigate('Settings')}

          >
            <Text style={styles.footerButtonText}>Settings</Text>
          </TouchableOpacity>
          
          <TouchableOpacity 
            style={styles.footerButton}
            onPress={() => navigation.navigate('Help')}
          >
            <Text style={styles.footerButtonText}>Help & Support</Text>
          </TouchableOpacity>
        </View>
      </DrawerContentScrollView>
    </LinearGradient>
  );
};

export default CustomDrawerContent;

const styles = StyleSheet.create({
  gradient: {
    flex: 1,
  },
  scrollView: {
    flex: 1,
    paddingTop: hp(4),
  },
  header: {
    alignItems: 'center',
    paddingBottom: hp(3),
  },
  logoContainer: {
    marginBottom: hp(1),
  },
  logoBackground: {
    width: wp(15),
    height: wp(15),
    borderRadius: wp(7.5),
    backgroundColor: '#e74c3c',
    justifyContent: 'center',
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  logoText: {
    fontSize: wp(8),
    color: 'white',
    fontWeight: 'bold',
  },
  appName: {
    fontSize: wp(5.5),
    fontWeight: 'bold',
    color: 'white',
    marginBottom: hp(0.5),
  },
  tagline: {
    fontSize: wp(3.5),
    color: '#ccc',
    fontStyle: 'italic',
  },
  divider: {
    height: 1,
    backgroundColor: 'rgba(255,255,255,0.1)',
    marginHorizontal: wp(5),
    marginBottom: hp(3),
  },
  menuContainer: {
    flex: 1,
    paddingHorizontal: wp(5),
  },
  menuItem: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingVertical: hp(1.8),
    marginBottom: hp(1.5),
    borderRadius: 12,
    backgroundColor: 'rgba(255,255,255,0.08)',
    paddingHorizontal: wp(4),
  },
  iconContainer: {
    width: wp(8),
    height: wp(8),
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: wp(4),
    backgroundColor: 'rgba(255,255,255,0.1)',
    marginRight: wp(3),
  },
  icon: {
    fontSize: wp(4),
  },
  menuText: {
    color: 'white',
    fontSize: wp(4.2),
    fontWeight: '600',
    letterSpacing: 0.5,
  },
  footer: {
    marginTop: 'auto',
    paddingHorizontal: wp(5),
    paddingBottom: hp(3),
  },
  footerButton: {
    paddingVertical: hp(1.5),
    marginTop: hp(1),
    borderRadius: 8,
    backgroundColor: 'rgba(255,255,255,0.05)',
    alignItems: 'center',
  },
  footerButtonText: {
    color: '#ccc',
    fontSize: wp(3.8),
  },
});