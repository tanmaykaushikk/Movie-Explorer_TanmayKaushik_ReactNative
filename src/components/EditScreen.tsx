import {
  Dimensions,
  ImageBackground,
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
  View,
  ScrollView,
  Image,
  Alert,
} from "react-native";
import React, { useState } from "react";
import LinearGradient from "react-native-linear-gradient";
import { launchImageLibrary, ImageLibraryOptions, Asset } from "react-native-image-picker";
import { createMovie } from "../utils/Api";
import { useRoute, useNavigation } from "@react-navigation/native";
import Toast from "react-native-toast-message";

const { height, width } = Dimensions.get("window");

const wp = (percent: number) => (width * percent) / 100;
const hp = (percent: number) => (height * percent) / 100;

const EditScreen = () => {
  const route = useRoute();
  const { id } = (route.params || {}) as { id: number };
  const navigation = useNavigation();
  const [invalidFields, setInvalidFields] = useState<Set<string>>(new Set());

  const [form, setForm] = useState({
    title: "",
    genre: "",
    release_year: "",
    director: "",
    duration: "",
    description: "",
    main_lead: "",
    streaming_platform: "",
    rating: "",
    posterPreview: "",
    posterFile: null as any,
  });

  const handleChange = (key: string, value: string) => {
    setForm((prev) => ({ ...prev, [key]: value }));

    if (invalidFields.has(key)) {
      setInvalidFields((prev) => {
        const update = new Set(prev);
        return update;
      });
    }
  };

  const pickImage = async () => {
    const options: ImageLibraryOptions = {
      mediaType: "photo",
      quality: 1,
      selectionLimit: 1,
    };

    launchImageLibrary(options, (response) => {
      if (response.didCancel) {
        console.log("User cancelled image picker");
      } else if (response.errorCode) {
        console.error("ImagePicker Error: ", response.errorMessage);
        // Alert.alert("Error picking image");
        Toast.show({
          type: 'error',
          text1: 'Image Picker Error',
          text2: 'Something went wrong while picking image.',
        });
      } else if (response.assets && response.assets.length > 0) {
        const image: Asset = response.assets[0];
        const fileUri = image.uri || "";
        const fileName = fileUri.split("/").pop() || "image.jpg";
        const fileType = fileName.split(".").pop();

        const file = {
          uri: fileUri,
          name: fileName,
          type: image.type || "image/jpeg",
        };

        setForm((prev) => ({
          ...prev,
          posterPreview: fileUri,
          posterFile: file,
        }));
      }
    });
  };

  const handleAdd = async () => {
    const requiredFields = [
      "title",
      "genre",
      "release_year",
      "director",
      "duration",
      "description",
      "main_lead",
      "streaming_platform",
      "rating",
      "posterFile",

    ];

    const missingFields = requiredFields.filter((key) => {
      if (key === "posterFile") {
        return !form.posterFile;
      }
      return !form[key].trim();
    });

    if (missingFields.length > 0) {
      setInvalidFields(new Set(missingFields));
      // Alert.alert("Validation Error", "Please fill all required fields.");
      Toast.show({
        type: 'error',
        text1: 'Validation Error',
        text2: 'Please fill all required fields.',
      });
      return;
    }


    setInvalidFields(new Set());

    try {
      const movieFormData = {
        title: form.title,
        genre: form.genre,
        release_year: form.release_year,
        director: form.director,
        duration: form.duration,
        description: form.description,
        main_lead: form.main_lead,
        streaming_platform: form.streaming_platform,
        rating: form.rating,
        isPremium: false,
        poster: form.posterFile,
        banner: null,
      };

      const newMovie = await createMovie(movieFormData as any);
      if (newMovie) {
        // Alert.alert("Movie added successfully!");
        Toast.show({
          type: 'success',
          text1: 'Success',
          text2: 'Movie added successfully!',
        });
        setForm({
          title: "",
          genre: "",
          release_year: "",
          director: "",
          duration: "",
          description: "",
          main_lead: "",
          streaming_platform: "",
          rating: "",
          posterPreview: "",
          posterFile: null,
        });
      }
    } catch (error) {
      console.error("Add Movie Error:", error);
      // Alert.alert("Failed to add movie");
      Toast.show({
        type: 'error',
        text1: 'Error',
        text2: 'Failed to add movie',
      });
    }
  };

  return (
    <ImageBackground
      source={require("../assets/Images/loginbackground.jpg")}
      style={styles.background}
      resizeMode="cover"
    >
      <LinearGradient
        colors={["rgba(108, 115, 118, 0.83)", "rgb(1, 25, 35)"]}
        style={styles.gradient}
      >
        <ScrollView contentContainerStyle={styles.scrollContainer} showsVerticalScrollIndicator={false}>
          <Text style={styles.heading} testID="edit-screen-heading" >Edit Movie Details</Text>

          {[
            { key: "title", placeholder: "Title" },
            { key: "genre", placeholder: "Genre" },
            { key: "release_year", placeholder: "Release Year", keyboardType: "numeric" },
            { key: "director", placeholder: "Director" },
            { key: "duration", placeholder: "Duration (e.g., 2h 15m)" },
            { key: "description", placeholder: "Description", multiline: true },
            { key: "main_lead", placeholder: "Main Lead Actor" },
            { key: "streaming_platform", placeholder: "Streaming Platform" },
            { key: "rating", placeholder: "Rating (e.g., 8.5)", keyboardType: "decimal-pad" },
          ].map((field) => {
            const isInvalid = invalidFields.has(field.key);
            return (
              <TextInput
                key={field.key}
                style={[styles.input, field.multiline && styles.multilineInput, isInvalid && styles.invalidInput,]}
                placeholder={field.placeholder}
                placeholderTextColor="silver"
                value={form[field.key]}
                onChangeText={(text) => handleChange(field.key, text)}
                keyboardType={field.keyboardType || "default"}
                multiline={field.multiline || false}
                testID={`input-${field.key}`}
              />
            );
          })}

          <TouchableOpacity onPress={pickImage} style={[styles.imagePicker, invalidFields.has("posterFile") && styles.invalidInput]} testID="image-picker-button">
            <Text style={styles.imagePickerText}>Pick Poster Image</Text>
            {form.posterPreview ? (
              <Image
                source={{ uri: form.posterPreview }}
                style={{ width: 100, height: 150, marginTop: 10 }}
                testID="poster-preview"
              />
            ) : null}
          </TouchableOpacity>

          <TouchableOpacity style={styles.button} onPress={handleAdd} testID="add-movie-button">
            <Text style={styles.buttonText}>Add Movie</Text>
          </TouchableOpacity>
        </ScrollView>
      </LinearGradient>
    </ImageBackground>
  );
};

export default EditScreen;

const styles = StyleSheet.create({
  background: {
    flex: 1,
  },
  gradient: {
    flex: 1,
    paddingHorizontal: wp(5),
    paddingTop: hp(5),
    justifyContent: "center",
  },
  scrollContainer: {
    paddingBottom: hp(10),
  },
  heading: {
    fontSize: wp(6),
    fontWeight: "bold",
    color: "white",
    textAlign: "center",
    marginBottom: hp(3),
  },
  input: {
    backgroundColor: "transparent",
    borderWidth: 1,
    borderColor: "#ccc",
    borderRadius: wp(2),
    paddingHorizontal: wp(4),
    fontSize: wp(4),
    marginBottom: hp(2),
    color: "white",
    height: hp(6),
  },
  multilineInput: {
    height: hp(12),
    textAlignVertical: "top",
  },
  invalidInput: {
    borderColor: "red",
  },
  imagePicker: {
    borderWidth: 1,
    borderColor: "#ccc",
    padding: 10,
    borderRadius: wp(2),
    alignItems: "center",
    marginBottom: hp(2),
  },
  imagePickerText: {
    color: "white",
    fontSize: wp(4),
  },
  button: {
    backgroundColor: "transparent",
    paddingVertical: hp(2),
    borderRadius: wp(2),
    alignItems: "center",
    borderWidth: 1,
    borderColor: "silver",
    marginTop: hp(2),
  },
  buttonText: {
    color: "white",
    fontSize: wp(4),
    fontWeight: "600",
  },
});

























































