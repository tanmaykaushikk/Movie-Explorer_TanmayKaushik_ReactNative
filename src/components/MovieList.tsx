import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  ScrollView,
  TouchableWithoutFeedback,
  Image,
  Dimensions,
  Alert,
} from "react-native";
import React from "react";
import { useNavigation } from "@react-navigation/native";
import { useState, useEffect } from "react";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { deleteMovie } from "../utils/Api";
import { getMoviesById } from "../utils/Api";

const { width } = Dimensions.get("window");

interface MovieItem {
  id: number;
  title: string;
  genre: string;
  release_year: number;
  rating: number;
  director: string;
  duration: number;
  description: string;
  premium: boolean;
  main_lead: string;
  streaming_platform: string;
  poster_url: string;
  banner_url: string;
}

interface MovieListProps {
  title: string;
  data: MovieItem[];
  handleClick: (item: MovieItem) => void;
  isAdmin: boolean;
  isPremiumSubscribed: boolean;
}

type NavigationParamList = {
  SeeAll: { title: string; movies: MovieItem[] };
  Movie: { movie: MovieItem };
  Update: { movie: MovieItem }; 
};

const MovieList: React.FC<MovieListProps> = ({ title, data, handleClick, isAdmin,isPremiumSubscribed }) => {
  const navigation = useNavigation<NativeStackNavigationProp<NavigationParamList>>();

  const [movieList , setMovieList] = useState<MovieItem[]>(data);

  useEffect(()=>{
    setMovieList(data);
  },[data]);

  const handleDelete = (id: number) => {
    Alert.alert(
      "Confirm Deletion",
      "Are you sure you want to delete this movie?",
      [
        {
          text: "Cancel",
          style: "cancel",
        },
        {
          text: "Delete",
          onPress: async () => {
            try {
              const response = await deleteMovie(id);
              if (response) {
                setMovieList((prevList) => prevList.filter((movie) => movie.id !== id));
                Alert.alert("Movie Deleted Successfully");
                // Optionally, refresh the movie list here
              } else {
                Alert.alert("Failed to delete movie");
              }
            } catch (error) {
              console.error("Delete Movie Error:", error);
              Alert.alert("An error occurred while deleting the movie.");
            }
          },
          style: "destructive",
        },
      ],
      { cancelable: true }
    );
  };

  return (
    <View style={styles.container}>
      <View style={styles.title}>
        <Text style={styles.titleText}>{title}</Text>
        <TouchableOpacity onPress={() => navigation.navigate("SeeAll", { title, movies: movieList })}>
          <Text style={styles.seeAll}>See All</Text>
        </TouchableOpacity>
      </View>

      <ScrollView horizontal showsHorizontalScrollIndicator={false} contentContainerStyle={{ paddingLeft: 10 }}>
        {movieList.map((item) => (
          <TouchableWithoutFeedback
            key={item.id}
            onPress={async () => {
              try{
                const updateMovie = await getMoviesById(item.id);
                if(updateMovie){
                  handleClick(updateMovie);
                }else{
                  Alert.alert("Error. Could not fetch the movie")
                }
              }catch(error){
                Alert.alert("Error. Something went wrong fetching movie")
              }
            }}
            onLongPress={() => isAdmin && handleDelete(item.id)}
              testID={`movie-touchable-${item.id}`}
          >
            <View style={styles.movieImageContainer}>
              <Image source={{ uri: item.poster_url }} testID="image" accessibilityRole="image" style={styles.movieImage} />

              {item.premium && !isAdmin && !isPremiumSubscribed && (
                <Image
                  source={require("../assets/Images/crown.png")}
                  style={styles.premiumIcon}
                  resizeMode="contain"
                  accessibilityRole="image"
                />
              )}

              {isAdmin && (
                <TouchableOpacity style={styles.edit} onPress={() => navigation.navigate("Update" , {movie:item})}
                  testID={`edit-icon-${item.id}`}>
                  <Image accessibilityRole="image" source={require("../assets/Images/pen.png")} style={styles.edit} />
                </TouchableOpacity>
              )}
            </View>
          </TouchableWithoutFeedback>
        ))}
      </ScrollView>
    </View>
  );
};

export default MovieList;

const styles = StyleSheet.create({
  container: {
    marginBottom: 20,
  },
  title: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    marginBottom: 12,
    paddingVertical: 8,
  },
  titleText: {
    color: "white",
    fontSize: 22,
    fontWeight: "bold",
    letterSpacing: 0.5,
  },
  seeAll: {
    color: "silver",
    fontSize: 12,
    fontWeight: "bold",
  },
  movieImageContainer: {
    marginRight: 15,
    width: width * 0.3,
    height: width * 0.45,
    borderRadius: 10,
    overflow: "hidden",
    backgroundColor: "white",
    justifyContent: "center",
    alignItems: "center",
  },
  movieImage: {
    width: 120,
    height: 170,
  },
  edit: {
    position: "absolute",
    top: 5,
    right: 5,
    width: 20,
    height: 20,
    padding: 4,
    borderRadius: 12,
    zIndex: 1,
  },
  premiumIcon: {
    position: "absolute",
    top: 8,
    left: 8,
    width: 24,
    height: 24,
    zIndex: 2,
  },
});


























