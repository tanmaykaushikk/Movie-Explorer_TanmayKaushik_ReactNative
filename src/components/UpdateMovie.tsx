import React, { useEffect, useState } from "react";
import {
  Dimensions,
  Text,
  TextInput,
  StyleSheet,
  ScrollView,
  Switch,
  Alert,
  View,
  TouchableOpacity,
  Image,
  ImageBackground,
} from "react-native";
import { useRoute, RouteProp, useNavigation } from "@react-navigation/native";
import LinearGradient from "react-native-linear-gradient";
import { getMoviesById, updateMovie } from "../utils/Api";
import Toast from "react-native-toast-message";

const { height, width } = Dimensions.get("window");
const wp = (percent: number) => (width * percent) / 100;
const hp = (percent: number) => (height * percent) / 100;

type NavigationParamList = {
  Update: { movie: MovieItem };
};

type MovieItem = {
  id: number;
  title: string;
  genre: string;
  release_year: number;
  rating: number;
  director: string;
  duration: number;
  description: string;
  premium: boolean;
  main_lead: string;
  streaming_platform: string;
  poster_url: any;
  banner_url: any;
};

const UpdateMovie = () => {
  const route = useRoute<RouteProp<NavigationParamList, "Update">>();
  const navigation = useNavigation();
  const { movie } = route.params;

  const [loading, setLoading] = useState(true);
  const [formState, setFormState] = useState({
    title: "",
    genre: "",
    release_year: "",
    rating: "",
    director: "",
    duration: "",
    description: "",
    main_lead: "",
    streaming_platform: "",
    isPremium: false,
    poster_url: "",
  });



  const fetchUpdateMovieData = async () => {
    setLoading(true);
    try {
      const data = await getMoviesById(movie?.id);
      setFormState({
        title: data?.title || "",
        genre: data?.genre || "",
        release_year: data?.release_year !== undefined ? data?.release_year.toString() : "",
        rating: data?.rating !== undefined ? data?.rating.toString() : "",
        director: data?.director || "",
        duration: data?.duration !== undefined ? data?.duration.toString() : "",
        description: data?.description || "",
        main_lead: data?.main_lead || "",
        streaming_platform: data?.streaming_platform || "",
        isPremium: !!data?.premium,
        poster_url: data?.poster_url || "",
      });
    } catch (err: any) {
      // Alert.alert("Error", "Failed to load movie data?");
      Toast.show({
        type: 'error',
        text1: 'Error',
        text2: 'Failed to load movie data',
      });
    } finally {
      setLoading(false);
    }
  };

  useEffect(() => {
    fetchUpdateMovieData();
  }, []);

  const handleChange = (key: keyof typeof formState, value: string | boolean) => {
    setFormState({ ...formState, [key]: value });
  };

  const handleUpdate = async () => {
    const payload = {
      ...formState,
      release_year: parseInt(formState.release_year),
      rating: parseFloat(formState.rating),
      duration: parseInt(formState.duration),
      premium: formState.isPremium,
      poster_url: formState.poster_url,
    };

    try {
      const result = await updateMovie(movie.id, payload);
      console.log("result", result)
      if (result) {
        // Alert.alert("Success", "Movie updated successfully");
        Toast.show({
          type: 'success',
          text1: 'Success',
          text2: 'Movie updated successfully',
        });
        navigation.goBack();
      } else {
        // Alert.alert("Error", "Failed to update movie");
        Toast.show({
          type: 'error',
          text1: 'Error',
          text2: 'Failed to update movie',
        })
      }
    } catch (error) {
      // Alert.alert("Error", "Something went wrong");
      Toast.show({
        type: 'error',
        text1: 'Error',
        text2: 'Something went wrong',
      });
    }
  };


  return (
    <ImageBackground
      source={require("../assets/Images/loginbackground.jpg")}
      style={styles.background}
      resizeMode="cover"
    >
      <LinearGradient colors={["rgba(108, 115, 118, 0.83)", "rgb(1, 25, 35)"]} style={styles.gradient}>
        <ScrollView contentContainerStyle={styles.scrollContainer} showsVerticalScrollIndicator={false}>
          <Text style={styles.heading}>Update Movie Details</Text>

          {[
            { key: "title", placeholder: "Title" },
            { key: "genre", placeholder: "Genre" },
            { key: "release_year", placeholder: "Release Year", keyboardType: "numeric" },
            { key: "director", placeholder: "Director" },
            { key: "duration", placeholder: "Duration (e.g., 120)", keyboardType: "numeric" },
            { key: "description", placeholder: "Description", multiline: true },
            { key: "main_lead", placeholder: "Main Lead Actor" },
            { key: "streaming_platform", placeholder: "Streaming Platform" },
            { key: "rating", placeholder: "Rating (e.g., 8.5)", keyboardType: "decimal-pad" },
          ].map((field) => (
            <TextInput
              key={field.key}
              style={[styles.input, field.multiline && styles.multilineInput]}
              placeholder={field.placeholder}
              placeholderTextColor="silver"
              value={formState[field.key]}
              onChangeText={(text) => handleChange(field.key, text)}
              keyboardType={field.keyboardType || "default"}
              multiline={field.multiline || false}
            />
          ))}

          <View style={styles.switchContainer}>
            <Text style={styles.label}>Premium</Text>
            <Switch
              value={formState.isPremium}
              onValueChange={(value) => handleChange("isPremium", value)}
            />
          </View>

          {formState.poster_url ? (
            <Image
              source={{ uri: formState.poster_url }}
              style={{ width: 100, height: 150, alignSelf: "center", marginBottom: 10 }}
            />
          ) : null}

          <TouchableOpacity style={styles.button} onPress={handleUpdate}>
            <Text style={styles.buttonText}>Update Movie</Text>
          </TouchableOpacity>
        </ScrollView>
      </LinearGradient>
    </ImageBackground>
  );
};

export default UpdateMovie;

const styles = StyleSheet.create({
  background: {
    flex: 1,
  },
  gradient: {
    flex: 1,
    paddingHorizontal: wp(5),
    paddingTop: hp(5),
    justifyContent: "center",
  },
  scrollContainer: {
    paddingBottom: hp(10),
  },
  heading: {
    fontSize: wp(6),
    fontWeight: "bold",
    color: "white",
    textAlign: "center",
    marginBottom: hp(3),
  },
  input: {
    backgroundColor: "transparent",
    borderWidth: 1,
    borderColor: "#ccc",
    borderRadius: wp(2),
    paddingHorizontal: wp(4),
    fontSize: wp(4),
    marginBottom: hp(2),
    color: "white",
    height: hp(6),
  },
  multilineInput: {
    height: hp(12),
    textAlignVertical: "top",
  },
  label: {
    color: "white",
    fontSize: wp(4),
    marginBottom: 8,
  },
  switchContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    borderWidth: 1,
    borderColor: "#ccc",
    borderRadius: wp(2),
    padding: 10,
    marginBottom: hp(2),
  },
  button: {
    backgroundColor: "transparent",
    paddingVertical: hp(2),
    borderRadius: wp(2),
    alignItems: "center",
    borderWidth: 1,
    borderColor: "silver",
    marginTop: hp(2),
  },
  buttonText: {
    color: "white",
    fontSize: wp(4),
    fontWeight: "600",
  },
});
