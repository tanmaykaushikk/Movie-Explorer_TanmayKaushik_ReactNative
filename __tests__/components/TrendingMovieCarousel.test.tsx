jest.mock('react-native-reanimated-carousel', () => {
  const React = require('react');
  const { View } = require('react-native');
  return ({ data, renderItem }: any) => (
    <View>{data.map((item: any, index: number) => renderItem({ item, index }))}</View>
  );
});

jest.mock('react-native-reanimated', () => require('react-native-reanimated/mock'));
jest.mock('react-native-gesture-handler', () => ({}));

import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import TrendingMoviesCarousel from "../../src/components/TrendingMovieCarousel";
import { itemProp } from "../../src/components/TrendingMovieCarousel";
import { NavigationContainer } from '@react-navigation/native';

const mockData: itemProp[] = [
  {
    id: 1,
    title: 'Test Movie',
    genre: 'Action',
    release_year: 2022,
    rating: 8.5,
    director: 'Test Director',
    duration: 120,
    description: 'Test Description',
    premium: true,
    main_lead: 'Test Lead',
    streaming_platform: 'Test Platform',
    poster_url: 'https://example.com/poster.jpg',
    banner_url: 'https://example.com/banner.jpg',
  },
];

const renderWithNavigation = (ui: React.ReactElement) => {
  return render(<NavigationContainer>{ui}</NavigationContainer>);
};

describe('TrendingMoviesCarousel Component', () => {
  it('renders "Trending" text', () => {
    const { getByText } = renderWithNavigation(
      <TrendingMoviesCarousel
        data={mockData}
        isAdmin={false}
        handleClick={jest.fn()}
      />
    );
    expect(getByText('Trending')).toBeTruthy();
  });

  it('renders poster image', () => {
    const { getByRole } = renderWithNavigation(
      <TrendingMoviesCarousel
        data={mockData}
        isAdmin={false}
        handleClick={jest.fn()}
      />
    );
    const image = getByRole('image');
    expect(image.props.source.uri).toBe(mockData[0].poster_url);
  });

  it('shows premium icon if movie is premium', () => {
    const { getByTestId } = renderWithNavigation(
      <TrendingMoviesCarousel
        data={mockData}
        isAdmin={false}
        handleClick={jest.fn()}
      />
    );
    expect(getByTestId('premium-icon')).toBeTruthy();
  });

  it('shows edit icon if user is admin', () => {
    const { getByTestId } = renderWithNavigation(
      <TrendingMoviesCarousel
        data={mockData}
        isAdmin={true}
        handleClick={jest.fn()}
      />
    );
    expect(getByTestId('edit-icon')).toBeTruthy();
  });

  it('calls handleClick when card is pressed', () => {
    const mockHandleClick = jest.fn();

    const { getByTestId } = renderWithNavigation(
      <TrendingMoviesCarousel
        data={mockData}
        isAdmin={false}
        handleClick={mockHandleClick}
      />
    );

    fireEvent.press(getByTestId(`card-${mockData[0].id}`));
    expect(mockHandleClick).toHaveBeenCalledWith(mockData[0]);
  });

  // âœ… EXTRA TEST CASES TO INCREASE COVERAGE

  it('does not show premium icon if movie is not premium', () => {
    const nonPremiumData = [{ ...mockData[0], premium: false }];
    const { queryByTestId } = renderWithNavigation(
      <TrendingMoviesCarousel
        data={nonPremiumData}
        isAdmin={false}
        handleClick={jest.fn()}
      />
    );
    expect(queryByTestId('premium-icon')).toBeNull();
  });

  it('does not show edit icon if user is not admin', () => {
    const { queryByTestId } = renderWithNavigation(
      <TrendingMoviesCarousel
        data={mockData}
        isAdmin={false}
        handleClick={jest.fn()}
      />
    );
    expect(queryByTestId('edit-icon')).toBeNull();
  });

  it('renders nothing if data is empty', () => {
    const { queryByRole } = renderWithNavigation(
      <TrendingMoviesCarousel
        data={[]}
        isAdmin={false}
        handleClick={jest.fn()}
      />
    );
    expect(queryByRole('image')).toBeNull();
  });
});
