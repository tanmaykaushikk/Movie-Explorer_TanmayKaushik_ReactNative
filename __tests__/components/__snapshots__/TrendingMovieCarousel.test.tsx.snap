// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrendingMoviesCarousel Component handles empty data gracefully 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Trending
  </Text>
  <View />
</View>
`;
